### ISSUE TITLE
Cannot read properties of null (reading 'promise')

### ISSUE THREAD

**Issue by aznn** [qzind/tray#866]
I'm using qz-tray JS build `2.1.2` and I'm observing this error consistently in production environment. 

Stack Trace:
```
TypeError: Cannot read properties of null (reading 'promise')
  at WebSocket.e.websocket.connection.onclose(./node_modules/qz-tray/qz-tray.js:195:54)
```

From the stack trace it looks like `_qz.websocket.connection` is already null by the time `onclose` is called. 

I've not been able to reproduce it locally but In my user base it is observed few hundred times everyday (as reported on Sentry). Attaching sentry data which might be helpful.

![CleanShot 2021-09-17 at 14 30 45](https://user-images.githubusercontent.com/1500309/133755669-7618b76f-0bac-420e-b092-2ce3b93c3329.png)

**Comment by aznn** [comment]
I can see some fixes have been done for this in the latest js version. Closing this issue for now. I'll reopen if I observe the same issue on latest build.

**Comment by tresf** [comment]
@aznn thanks.  Yes, there was a race condition #751, fixed in #754.  If you find another, please let us know! :)
